templates/deployment.yml

```diff
@@ -29,11 +29,25 @@ spec:
       - image: armory/agent-k8s:{{ .Chart.AppVersion }}
         imagePullPolicy: IfNotPresent
         name: agent
-        {{- if .Values.cloudEnabled }}
         env:
+        {{- if .Values.cloudEnabled }}
         - name: ARMORY_HUB
           value: "true"
         {{- end }}
+        {{- if .Values.proxy }}
+          {{- with .Values.proxy.http_proxy }}
+        - name: HTTP_PROXY
```

Comment:
What if we expose all `env` section in `values.yml`? So we're not changing the chart for every possible env var.

Reply:
Is that the interface that we want? The fact that they're environment variables are more or less implementation details to the user. With our current config we're giving semantic meaning to "cloudEnabled" and "proxy" settings that allow us to change how we provide them to the manifests.

If on the other hand, we _do_ want to expose this to the user, then yeah it totally makes sense to just have a generic env section exposed.

WDYT?

Reply:
A drawback of helm charts is that people don't have full control over the manifests, and there are always cases when someone wants to change it in some way or another, that's the reason Halyard failed, people wanted to customize in every possible way the generated spinnaker manifests, so there were changes for exposing labels, then annotations, then sidecars, etc. I think exposing the full `env` section protect us at least from some future changes when some customer wants some new variable in their setup.

We can still provide semantic and have things like "cloudEnabled" that hide the implementation detail that this is an env var. How about a base `values.yml` like this (people can add more env vars as desired):
```
grpcUrl:  agents.cloud.armory.io:443
mode: agent
 cloudEnabled: true
 kubernetes:
   accounts:
   - name: account1

env:
  - name: HTTP_PROXY
    value: ""
  - name: HTTPS_PROXY
    value: ""
  - name: NO_PROXY
    value: ""
```

During this sprint we expect to expose the full config file, so ideally in the end we have something like this, given that helm allows us to derive if we're running with cloud enabled just by looking at the config. Tthis will be a breaking change for people using the current helm chart, but will allow them to actually use _all_ the config settings we're implementing in the service:

```
config:
  kubernetes:
    accounts:
    - name: rogue
  hub:
    connection:
      grpc: agents.cloud.armory.io:443

env: []
```

Reply:
Thanks for explaining that @german-muzquiz! Given the context you've shared I agree that the env approach is better for the time being. I'll update that now!